src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -100,8 +100,19 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			{
 				AlterTableCmd *command = (AlterTableCmd *) lfirst(commandCell);
 
-				if (command->subtype == AT_AddConstraint ||
-					command->subtype == AT_DropConstraint)
+				if (command->subtype == AT_AddConstraint)
+				{
+					Constraint *con = castNode(Constraint, command->def);
```

Comment:
`con` -> `constraint`. 

For consistency with the code below, maybe use `(Constrant *) command->def;`